### PR TITLE
feat(toolbox): Port toolbox to work with gateway

### PR DESCRIPTION
**Pull request by StanislavBreadless**
## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

File: zk_toolbox/crates/zk_inception/src/commands/chain/migrate_to_gateway.rs [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+    if receipt.status == Some(U64::from(1)) {
+        println!("Transaction completed successfully!");
+    } else {
+        panic!("Transaction failed!");
+    }
+
+    Ok(())
+}
+
+async fn call_script(
```
Comment (kelemeno) [inline zk_toolbox/crates/zk_inception/src/commands/chain/migrate_to_gateway.rs:406]: This function is duplicated, probably ok for now

File: .github/workflows/ci-core-reusable.yml [hunk 2]
```diff
@@ -318,45 +367,6 @@ jobs:
       #     ci_run sleep 10
       #     ci_run zk test i upgrade
 
-      - name: Show server.log logs
```
Comment (kelemeno) [inline .github/workflows/ci-core-reusable.yml:321]: Do we not want to show logs? Probably something else is displaying them

File: core/lib/dal/src/eth_sender_dal.rs [hunk 3]
```diff
@@ -746,6 +750,43 @@ impl EthSenderDal<'_, '_> {
 
         Ok(())
     }
+
+    pub async fn get_batch_finalization_info(
```
Comment (kelemeno) [inline core/lib/dal/src/eth_sender_dal.rs:754]: I don't understand what happens here?

File: .github/workflows/ci-core-reusable.yml (hunk 2 above)
Comment (StanislavBreadless) [reply .github/workflows/ci-core-reusable.yml:321]: yeah, the Upload logs action publishes all the logs to github and can be accessible via a link (very helpful when we'll run tests for multiple chains at the same time)

File: zk_toolbox/crates/zk_inception/src/commands/chain/migrate_to_gateway.rs (hunk 1 above)
Comment (StanislavBreadless) [reply zk_toolbox/crates/zk_inception/src/commands/chain/migrate_to_gateway.rs:406]: let's keep it this way for now, I have a feeling that Danil would want to refactor lots of stuff here anywhere

**Review by kelemeno — APPROVED** [review]

File: zk_toolbox/crates/common/src/forge.rs [hunk 4]
```diff
@@ -69,6 +69,17 @@ impl ForgeScript {
                 return Ok(res?);
             }
         }
+
+        // TODO: This line is very helpful for debugging purposes,
+        // maybe it makes sense to make it conditionally displayed.
+        let command = format!(
```
Comment (kelemeno) [inline zk_toolbox/crates/common/src/forge.rs:75]: not a big issue, but small code duplication here as well

**Pull request merged**